models/nft/cryptopunks_trades.sql

```diff
@@ -0,0 +1,100 @@
+with cryptopunksmarket_evt_punkbought as (
+  select *
+  from {{ ref('cryptopunks_CryptoPunksMarket_evt_PunkBought') }}
+),
+
+cryptopunksmarket_evt_punkbidentered as (
+  select *
+  from {{ ref('cryptopunks_CryptoPunksMarket_evt_PunkBidEntered') }}
+),
+
+erc20_token_transfers as (
+  select
+    evt_tx_hash,
+    `from` as from_address,
+    `to` as to_address
+  from {{ ref('ERC20_evt_Transfer') }}
+  where dt >= '{{ var("start_ts") }}'
+    and dt < '{{ var("end_ts") }}'
+    and contract_address = '0xb47e3cd837ddf8e4c57f05d70ab865de6e193bbb'
+),
+
+address_info as (
+  select
+    a.dt,
+    a.evt_block_time as block_time,
+    a.evt_block_number as block_number,
+    a.evt_tx_hash as tx_hash,
+    a.punkIndex as nft_token_id,
+    a.fromAddress as from_address,
+    a.value,
+    case
+      when a.toaddress = '0x0000000000000000000000000000000000000000' then b.to_address else a.toaddress
+    end as to_address
+  from cryptopunksmarket_evt_punkbought a
+  left join erc20_token_transfers b
+    on a.evt_tx_hash = b.evt_tx_hash and a.fromAddress = b.from_address
+),
+
+punk_trade as (
+  select
+    dt,
+    block_time,
+    block_number,
+    tx_hash,
+    nft_token_id,
+    from_address,
+    to_address,
+    (case when value = 0 then coalesce(bid_value, 0) else value end) as original_currency_amount
+  from (
+    select
+      a.*,
+      b.value as bid_value,
+      row_number()over(partition by a.nft_token_id, a.block_time order by b.evt_block_time desc) as rank
+    from address_info a
+    left join cryptopunksmarket_evt_punkbidentered b
+      on a.nft_token_id = b.punkIndex and b.fromAddress = a.to_address and b.evt_block_time <= a.block_time
+  )
+  where rank = 1 and value < 20000 * power(10, 18)
```

Comment:
What's the meaning of 20000?

Reply:
Many dune dashboards add this filter, I found just filtered the transaction of #9998 (124457ETH, the money from this NFT trade ended up right back where it started）。
Maybe used in the dashboard to filter outliers,  I'd remove this filter here.

Reply:
OKay. I guess the filter is reasonable. Maybe we can just filter by transaction id and add some comments to it.